project/Scope.rst

```diff
@@ -0,0 +1,31 @@
+Scope of Halium
+===============
+
+Halium's scope primarily consists of 2 parts:
+
+1. Halium's Android is a minimal Android distribution designed to run within an LXC container. It will provide a set of interfaces to let the host of the LXC container utilize Android's HAL to communicate with the hardware of Android-based smartphones. These interfaces come in 3 forms:
+
+    1. Native Android interfaces. (e.g. rild's socket, Audio HAL (dlopen'ed directly), (original) WiFi HAL (`a standard libhybris wrapper <https://github.com/libhybris/libhybris/tree/master/hybris/wifi>`_))
+    2. Libhybris compat layer. Usually used to wrap Android C++ interfaces, contains some logic, and has its counter-part compiled in the Android tree. (e.g. `HWC2 compat layer <https://github.com/libhybris/libhybris/tree/master/compat/hwc2>`_)
+    3. Custom interfaces. This will contain more than a compat layer and may have quite deep integration within Android code. (e.g. Ubuntu Touch's camera interfaces, which consists of `camera compat layer <https://github.com/libhybris/libhybris/tree/master/compat/camera>`_, `custom media compat layer <https://github.com/libhybris/libhybris/tree/master/compat/media>`_, and `deep modifications <https://github.com/ubports/android_frameworks_av/compare/1516fdb...c66dd36>`_ of Android's libmedia). If possible, we would like to reduce this kind of interface as much as possible to reduce our workload, especially those doing deep modification to Android.
```

Comment:
Instead of linking to the commits, potentially explain nature of that changes?

Reply:
The phrase "deep modifications to Android's libmedia" should describe it. So, I shouldn't link to the commits? Or I should describe it in more detail?

Reply:
Please see the footnote at the bottom of the page.